# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/gets.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

This implements gets for "minised".

  - Rene Rebe <rene@exactcode.de>

--- /dev/null	1970-01-01 01:00:00.000000000 +0100
+++ dietlibc-0.28/libstdio/gets.c	2005-02-02 16:41:10.406257872 +0100
@@ -0,0 +1,19 @@
+#include "dietstdio.h"
+#include "dietwarning.h"
+
+char *gets_unlocked(char *s) {
+  char *orig=s;
+  register int c;
+  for (;;) {
+    c=fgetc_unlocked(stdin);
+    if (c==EOF || c=='\n') break;
+    *s++=c;
+  }
+  if (c==EOF || ferror_unlocked(stdin))
+    return 0;
+  *s=0;
+  return orig;
+}
+
+char*gets(char*s) __attribute__((weak,alias("gets_unlocked")));
+link_warning("gets","warning: Avoid gets, it is insecure.")
